src/nbl/video/CVKLogicalDevice.h

```diff
@@ -178,52 +802,382 @@ class CVKLogicalDevice final : public ILogicalDevice
+        }
+
+        // Should just do this check in ISPIRVOptimizer::optimize
+        if (!spirv)
+            return nullptr;
+
+        if (spvopt)
+            spirv = spvopt->optimize(spirv.get(), m_physicalDevice->getDebugCallback()->getLogger());
+
+        if (!spirv)
+            return nullptr;
+
+        VkShaderModuleCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_SHADER_MODULE_CREATE_INFO };
+        vk_createInfo.pNext = nullptr;
+        vk_createInfo.flags = static_cast<VkShaderModuleCreateFlags>(0); // reserved for future use by Vulkan
+        vk_createInfo.codeSize = spirv->getSize();
+        vk_createInfo.pCode = reinterpret_cast<const uint32_t*>(spirv->getPointer());
+
+        VkShaderModule vk_shaderModule;
+        if (vkCreateShaderModule(m_vkdev, &vk_createInfo, nullptr, &vk_shaderModule) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<video::CVulkanSpecializedShader>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), vk_shaderModule, shaderStage);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
     core::smart_refctd_ptr<IGPUBufferView> createGPUBufferView_impl(IGPUBuffer* _underlying, asset::E_FORMAT _fmt, size_t _offset = 0ull, size_t _size = IGPUBufferView::whole_buffer) override
     {
-        return nullptr;
+        if (_underlying->getAPIType() != EAT_VULKAN)
+            return nullptr;
+
+        VkBuffer vk_buffer = static_cast<const CVulkanBuffer*>(_underlying)->getInternalObject();
+
+        VkBufferViewCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_BUFFER_VIEW_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // pNext must be NULL
+        vk_createInfo.flags = static_cast<VkBufferViewCreateFlags>(0); // flags must be 0
+        vk_createInfo.buffer = vk_buffer;
+        vk_createInfo.format = getVkFormatFromFormat(_fmt);
+        vk_createInfo.offset = _offset;
+        vk_createInfo.range = _size;
+
+        VkBufferView vk_bufferView;
+        if (vkCreateBufferView(m_vkdev, &vk_createInfo, nullptr, &vk_bufferView) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanBufferView>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this),
+                core::smart_refctd_ptr<IGPUBuffer>(_underlying), _fmt, _offset,
+                _size, vk_bufferView);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
     core::smart_refctd_ptr<IGPUImageView> createGPUImageView_impl(IGPUImageView::SCreationParams&& params) override
     {
-        return nullptr;
+        VkImageViewCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkImageViewASTCDecodeModeEXT, VkImageViewUsageCreateInfo, VkSamplerYcbcrConversionInfo, VkVideoProfileKHR, or VkVideoProfilesKHR
```

Comment:
`VkImageViewUsageCreateInfo` sounds interesting, what does it do and what's its purpose?

Reply:
@Erfan-Ahmadi a summary please.